ceph/ceph/templates/bin/_osd_disk_prepare.sh.tpl

```diff
@@ -18,30 +18,26 @@ function osd_disk_prepare {
   fi
   timeout 10 ceph ${CLI_OPTS} --name client.bootstrap-osd --keyring $OSD_BOOTSTRAP_KEYRING health || exit 1
 
-  # check device status first
-  if ! parted --script ${OSD_DEVICE} print > /dev/null 2>&1; then
-    if [[ ${OSD_FORCE_ZAP} -eq 1 ]]; then
-      log "It looks like ${OSD_DEVICE} isn't consistent, however OSD_FORCE_ZAP is enabled so we are zapping the device anyway"
-      ceph-disk -v zap ${OSD_DEVICE}
-    else
-      log "Regarding parted, device ${OSD_DEVICE} is inconsistent/broken/weird."
-      log "It would be too dangerous to destroy it without any notification."
-      log "Please set OSD_FORCE_ZAP to '1' if you really want to zap this disk."
-      exit 1
-    fi
-  fi
-
-  # then search for some ceph metadata on the disk
+  # search for some ceph metadata on the disk
   if [[ "$(parted --script ${OSD_DEVICE} print | egrep '^ 1.*ceph data')" ]]; then
-    if [[ ${OSD_FORCE_ZAP} -eq 1 ]]; then
-      log "It looks like ${OSD_DEVICE} is an OSD, however OSD_FORCE_ZAP is enabled so we are zapping the device anyway"
-      ceph-disk -v zap ${OSD_DEVICE}
-    else
-      log "INFO- It looks like ${OSD_DEVICE} is an OSD, set OSD_FORCE_ZAP=1 to use this device anyway and zap its content"
-      log "You can also use the zap_device scenario on the appropriate device to zap it"
-      log "Moving on, trying to activate the OSD now."
+    log "It looks like ${OSD_DEVICE} is already an OSD"
+    log "Checking if it belongs to this cluster"
+    tmp_osd_mount="/var/lib/ceph/tmp/`echo $RANDOM`/"
+    mkdir -p $tmp_osd_mount
+    mount ${OSD_DEVICE}1 ${tmp_osd_mount}
+    osd_cluster_fsid=`cat ${tmp_osd_mount}/ceph_fsid`
+    umount ${tmp_osd_mount} && rmdir ${tmp_osd_mount}
+    cluster_fsid=`ceph ${CLI_OPTS} --name client.bootstrap-osd --keyring $OSD_BOOTSTRAP_KEYRING fsid`
+    if [ "${osd_cluster_fsid}" == "${cluster_fsid}" ]; then
+      log "The OSD on ${OSD_DEVICE} belongs to this cluster, moving to activate phase"
       return
     fi
+    log "This OSD belonged to another cluster: ${osd_cluster_fsid}. Current cluster fsid: ${cluster_fsid}"
```

Comment:
do we not want to error-exit here?

Reply:
The idea is the drive belonged to another cluster, we can zap it. If we want to prevent that and force the user to manually zap it, I can change that.